apps/web/pages/gallery/[galleryId]/edit.tsx

```diff
@@ -100,6 +105,10 @@ function NewEditGalleryPageInner({ queryRef }: NewEditGalleryPageInnerProps) {
           hasUnsavedChanges={hasUnsavedChanges}
           onBack={handleBack}
           onDone={handleDone}
+          step={step}
+          dialogMessage={dialogMessage}
+          onNextStep={nextStep}
+          dialogOnClose={handleClose}
```

Comment:
so i noticed we're using `useOnboardingDialogContext` on this component, then simply prop-drilling down to the children (`NewEditGalleryPageInner -> EditGalleryNavbar -> GalleryTitleSection`)

the convenience with React Context is that we can use the hooks anywhere within the tree, which means we can avoid prop drilling and only use the hook as close to the leaf nodes as possible.

ideally we should call `useOnboardingDialogContext` on `GalleryTitleSection.tsx`, and avoid passing props all the way down!

![image](https://user-images.githubusercontent.com/12162433/217372219-44f442aa-dbc3-4a68-86c6-02b13204c857.png)

Reply:
I think so too. 

But the way how our navbar works, they can't access the context because its higher in the DOM tree. 

Will post the detail explanation in slack